lib/config/mysql_config.go

```diff
@@ -22,138 +26,70 @@ type generalConfig struct {
 }
 
 //NewMysqlConfig generates and returns a new mysql config provider
-func NewMysqlConfig(address string, username string, password string, database string) (Provider, error) {
-	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s)/%s", username, password, address, database))
+func NewMysqlConfig(address, username, password, database, configPath string) (Provider, error) {
+	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s)/?multiStatements=true", username, password, address))
 	if err != nil {
 		return nil, err
 	}
-	provisioner := mysqlConfig{}
-	provisioner.db = db
-	return &provisioner, nil
+	return &mysqlConfig{
+		db:         db,
+		dbName:     database,
+		configPath: configPath,
+	}, nil
 }
 
-func (c *mysqlConfig) LoadConfiguration() (*Config, error) {
-	var configuration Config
-	var mgmt ManagementAPI
-	configuration.ManagementAPI = &mgmt
-	result, err := c.db.Query("SELECT * FROM Config")
+func (c *mysqlConfig) InitializeConfiguration() error {
+	if c.db == nil {
+		return fmt.Errorf("Database connection not opened")
+	}
+
+	// Create the database if it doesn't exist
+	if _, err := c.db.Exec("CREATE SCHEMA IF NOT EXISTS ? DEFAULT CHARACTER SET utf8", c.dbName); err != nil {
```

Comment:
Didn't know `create schema` is synonymous with `create database`. MySQL is terrible :(